extensions/wikia/ImageReview/ImageListGetter.class.php

```diff
@@ -0,0 +1,384 @@
+		}
+
+		if ( !$imagePage instanceof GlobalTitle ) {
+			return [ 'error' => 'Page does not exist' ];
+		}
+
+		if ( $imagePage->isRedirect() ) {
+			return [ 'error' => 'Page is a redirect' ];
+		}
+
+		$imageGlobalFile = new GlobalFile( $imagePage );
+		if ( !$imageGlobalFile->exists() ) {
+			return [ 'error' => 'File does not exist' ];
+		}
+
+		return [
+			'file' => $imageGlobalFile,
+		    'page' => $imagePage,
+		];
+	}
+
+	private function moveImagesToIcoStateFromWhere( array $where ) {
+		$count = count( $where );
+		if ( $count <= 0 ) {
+			return;
+		}
+
+		$values = [ 'state' => ImageStates::ICO_IMAGE ];
+		$where = [ implode( 'OR', $where ) ];
+
+		$this->imageStateUpdater->updateBatchImages( $values, $where );
+
+		WikiaLogger::instance()->info( "ImageReviewLog", [
+			'method' => __METHOD__,
+			'message' => "Updated {$count} images (type 'icons')",
+		] );
+	}
+
+	/**
+	 * When assigning images to a user, we move the images from the
+	 * "unreviewed" state, to the "reviewed" state. Which new state
+	 * depends on where the image is in the review process.
+	 *
+	 * If an image hasn't been reviewed at all, it moves from "unreviewed"
+	 * to "in review". If the image has been marked as rejected or questionable
+	 * during the first pass of reviews, it moves to "rejected in review"
+	 * or "questionable in review"
+	 * @return int
+	 */
+	private function getNewState() : int {
+		if ( $this->state == ImageStates::REJECTED ) {
+			return ImageStates::REJECTED_IN_REVIEW;
+		}
+
+		if ( $this->state == ImageStates::QUESTIONABLE ) {
+			return ImageStates::QUESTIONABLE_IN_REVIEW;
+		}
+
+		return ImageStates::IN_REVIEW;
+	}
+
+	private function fetchedEnoughImages() : bool {
+		return count( $this->imageList ) == self::LIMIT_IMAGES;
+	}
+
+	/**
+	 * Setup the labels used to indicated different states for each image
+	 * @return array
+	 */
+	private function getLabelValues() {
+		$labels = [];
+
+		$labels['labelOk'] = [
+			'value' => ImageStates::APPROVED,
+			'checked' => in_array( $this->state, [ImageStates::APPROVED, ImageStates::IN_REVIEW, ImageStates::UNREVIEWED] ) ? 'checked' : ''
+		];
+
+		$labels['labelDelete'] = [
+			'value' => $this->state == 'rejected' ? ImageStates::DELETED : ImageStates::REJECTED,
+			'checked' => in_array( $this->state, [ImageStates::REJECTED] ) ? 'checked' : ''
+		];
+
+		$labels['labelQuestionable'] = [
+			'value' => ImageStates::QUESTIONABLE,
+			'checked' => in_array( $this->state, [ImageStates::QUESTIONABLE, ImageStates::QUESTIONABLE_IN_REVIEW] ) ? 'checked' : ''
+		];
+
+		return $labels;
+	}
+
+	/**
+	* Add additional markup for various flagged states
+	* @param integer $flag
+	* @return array
+	*/
+	private function getFlagsContent( $flag ) {
+		$flagsContent = [];
+
+		if( $flag & ImageListGetter::FLAG_SUSPICOUS_USER ) {
+			$flagsContent['flag_suspicous_user'] = [
+				'title' => "Flagged: Susicious user. Click to go to uploader's profile"
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Flagged: Susicious user. Click to go to uploader's profile" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "title" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)